File: 1528-KidsWithTheGreatestNumberOfCandies/1528-KidsWithTheGreatestNumberOfCandies.py
# Last updated: 7/30/2025, 1:02:16 PM
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        m = max(candies)
        return [ i+extraCandies>=m for i in candies]
        

File: 1528-KidsWithTheGreatestNumberOfCandies/1528-KidsWithTheGreatestNumberOfCandies.java
// Last updated: 7/30/2025, 1:02:14 PM
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = Integer.MIN_VALUE;
        for(int i=0; i<candies.length; i++){
            if(candies[i] > max){
                max = candies[i];
            }
        }
        List<Boolean> ans = new ArrayList<>();
        for(int i=0; i<candies.length; i++){
            ans.add(candies[i]+extraCandies >= max);
        }
        return ans;
    }
}